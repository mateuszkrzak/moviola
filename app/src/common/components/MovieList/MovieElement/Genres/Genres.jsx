import React from 'react';
import PropTypes from 'prop-types';

import styles from './Genres.scss';

class Genres extends React.Component {
  static defaultProps = {
    genres: [],
  };

  render() {
    return <span styleName="genre"> {this.props.genres.join(' & ')} </span>;
  }
}

Genres.propTypes = {
  genres: PropTypes.arrayOf(PropTypes.string),
};

export default Genres;
