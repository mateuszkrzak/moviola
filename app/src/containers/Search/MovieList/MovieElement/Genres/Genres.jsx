import React from 'react';
import PropTypes from 'prop-types';

import styles from './Genres.scss';

class Genres extends React.Component {
  static defaultProps = {
    genres: [],
  };

  formatGenres = genres => genres.join(' & ');

  render() {
    return <span styleName="genre"> {this.formatGenres(this.props.genres)} </span>;
  }
}

Genres.propTypes = {
  genres: PropTypes.arrayOf(PropTypes.string),
};

export default Genres;
