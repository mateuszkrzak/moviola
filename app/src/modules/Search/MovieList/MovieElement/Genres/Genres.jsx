import React from 'react';
import CSSModules from 'react-css-modules';
import PropTypes from 'prop-types';

import styles from './Genres.scss';

class Genres extends React.Component {
  static defaultProps = {
    genres: [],
  };

  formatGenres = genres => genres.join(' & ');

  render() {
    return <span styleName="genre"> {this.formatGenres(this.props.genres)} </span>;
  }
}

Genres.propTypes = {
  genres: PropTypes.arrayOf(PropTypes.string),
};

export default CSSModules(Genres, styles);
