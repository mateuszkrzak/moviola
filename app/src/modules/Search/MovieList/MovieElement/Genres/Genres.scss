.genre {
  text-transform: uppercase;
  display: block;
}
