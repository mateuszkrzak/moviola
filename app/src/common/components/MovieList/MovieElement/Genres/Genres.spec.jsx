import { shallow } from 'enzyme';
import React from 'react';
import Genres from './Genres';

test('Genres component renders genres', () => {
  const mockGenresProps = ['Test Genre 1', 'Test Genre 2'];
  const wrapper = shallow(<Genres {...mockGenresProps} />);
  expect(wrapper).toMatchSnapshot();
});
