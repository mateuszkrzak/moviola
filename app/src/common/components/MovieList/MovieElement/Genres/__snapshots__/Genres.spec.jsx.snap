// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genres component renders genres 1`] = `
<span
  styleName="genre"
>
   
  Test Genre 1 & Test Genre 2
   
</span>
`;
